test/relay.test.js

```diff
@@ -0,0 +1,323 @@
+'use strict';
+
+var chai = require('chai')
+  , expect = chai.expect
+  , resolver = require('../src/resolver')
+  , helper = require('./helper')
+  , sequelize = helper.sequelize
+  , Sequelize = require('sequelize')
+  , Promise = helper.Promise
+  , attributeFields = require('../src/attributeFields');
+
+import {
+  GraphQLString,
+  GraphQLInt,
+  GraphQLFloat,
+  GraphQLNonNull,
+  GraphQLBoolean,
+  GraphQLEnumType,
+  GraphQLList,
+  GraphQLObjectType,
+  GraphQLSchema,
+  graphql
+} from 'graphql';
+
+import {
+  sequelizeNodeInterface
+} from '../src/relay';
+
+import {
+  globalIdField,
+  connectionDefinitions,
+  connectionArgs,
+  connectionFromArray
+} from 'graphql-relay';
+
+function generateTask(id) {
+  return {
+    id: id,
+    name: Math.random().toString()
+  }
+}
+
+describe('relay', function () {
+  var User
+    , Task
+    , userType
+    , taskType
+    , taskConnection
+    , nodeInterface
+    , Project
+    , projectType
+    , userConnection
+    , nodeField
+    , schema;
+
+  before(function () {
+    sequelize.modelManager.models = [];
+    sequelize.models = {};
+    User = sequelize.define('user', {
+      name: {
+        type: Sequelize.STRING
+      }
+    }, {
+      timestamps: false
+    });
+
+    Task = sequelize.define('task', {
+      name: {
+        type: Sequelize.STRING
+      }
+    }, {
+      timestamps: false
+    });
+
+    Project = sequelize.define('name', {
+      name: {
+        type: Sequelize.STRING
+      }
+    }, {
+      timestamps: false
+    });
+
+    User.Tasks = User.hasMany(Task, {as: 'tasks'});
+    Project.Users = Project.hasMany(User, {as: 'users'});
+
+    var node = sequelizeNodeInterface(sequelize, {Project: projectType, User: userType, Task: taskType});
```

Comment:
the declarations of the types are hoisted to the top of the file and at that time they are set in memory as undefined. The references to the memory allocations are passed along in the object passed to the sequelizeNodeInterface function. However by the time the type resolver is ran, which is ran when a query comes into the graphql server, the types are defined in memory and thus there is no issue with undefined variables.